File: data/data.code.lua.client/AckRoleRandNameOk.lua
module("AckRoleRandNameOk", package.seeall)


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = AckRoleRandNameOk})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.uname = ""
end


function decode(self, pack)
	self.uname = pack:readString()
	return self
end

return AckRoleRandNameOk

File: examples/Laya2Ts/src/game/proto/AckRoleRandNameOk.ts
import Packet from '../net/Packet'


export default class AckRoleRandNameOk
{
	private _uname: string;


	constructor(packet?: Packet) {
		if (packet) {
			this._uname = packet.ReadString();
		}
	}


	public get uname(): string { return this._uname; }
	public set uname(value: string) { this._uname = value; }
}
